packages/aws-arch/scripts/common.ts

```diff
@@ -0,0 +1,29 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+export const GENERATED_DIR = path.resolve(__dirname, '..', 'src', 'generated');
+
+// Ensure the generated dir exists
+fs.ensureDirSync(GENERATED_DIR);
+
+export function logCount(title: string, total: number | object | any[] | Set<any>, count: number | object | any[] | Set<any>, delta: boolean = false): void {
+	total = parseCount(total);
+	count = parseCount(count);
+	if (delta) {
+		count = total - count;
+	}
+	console.info(`${title}: %i% (%i of %i)`, (count / total) * 100, count, total)
+}
+
+export function parseCount(value: number | object | any[] | Set<any>): number {
+	if (value instanceof Set) {
```

Comment:
convert these to if/else as I am not sure if a Set is also represented as an object and hence it may be possible to run multiple if conditions given we are not returning within the closure.